_studio/shared/src/cm_mem_copy.cpp

```diff
@@ -3164,6 +3165,59 @@ mfxStatus CmCopyWrapper::CopyMirrorVideoToVideoMemory(void *pDst, void *pSrc, mf
 
 #define CM_RGB_SUPPORTED_COPY_SIZE(ROI) (ROI.width <= CM_RGB_MAX_GPUCOPY_SURFACE_HEIGHT && ROI.height <= CM_RGB_MAX_GPUCOPY_SURFACE_HEIGHT )
 
+bool CmCopyWrapper::CheckSurfaceContinuouslyAllocated(mfxFrameSurface1 *surf)
+{
+    mfxFrameData* ptrFrameData = &surf->Data;
+
+    switch (surf->Info.FourCC)
+    {
+    case MFX_FOURCC_P010:
+    case MFX_FOURCC_P016:
+    case MFX_FOURCC_P210:
+    case MFX_FOURCC_NV16:
+    case MFX_FOURCC_NV12:
+        {
+            mfxU32 stride_in_bytes = ptrFrameData->PitchLow + ((mfxU32)ptrFrameData->PitchHigh << 16);
+
+            size_t totalBufferUPSize = stride_in_bytes * ALIGN32(surf->Info.Height)
+                + stride_in_bytes * surf->Info.Height / 2;
+
+            if (ptrFrameData->UV + stride_in_bytes * surf->Info.Height / 2 > ptrFrameData->Y + totalBufferUPSize)
+            {
+                return false;
+            }
+
+            break;
+        }
+
+    case MFX_FOURCC_YV12:
+        {
+            mfxU32 Y_stride_in_bytes = ptrFrameData->PitchLow + ((mfxU32)ptrFrameData->PitchHigh << 8);
+            mfxU32 VU_stride_in_bytes = Y_stride_in_bytes / 2;
+
+            size_t totalBufferUPSize = Y_stride_in_bytes * ALIGN32(surf->Info.Height)
+                + VU_stride_in_bytes * surf->Info.Height;
+
+            if (ptrFrameData->U + VU_stride_in_bytes * surf->Info.Height / 2 > ptrFrameData->Y + totalBufferUPSize)
+                return false;
+
+            break;
+        }
+    case MFX_FOURCC_RGBP:
+        {
+            mfxU32 stride_in_bytes = ptrFrameData->PitchLow + ((mfxU32)ptrFrameData->PitchHigh << 8);
+
+            size_t totalBufferUPSize = stride_in_bytes * surf->Info.Height * 3;
+
+            if (ptrFrameData->R + stride_in_bytes * surf->Info.Height > ptrFrameData->B + totalBufferUPSize)
```

Comment:
BGR order is assumed, but GBR  or ??B will pass as well. Again, doesn't work if planes were allocated separately.